### PR TITLE
web: add the location to button click analytics

### PR DESCRIPTION
**Pull request by nicks**
Hello @lizzthabet, @landism,

Please review the following commits I made in branch nicks/metrics:

551c4192c20c8c72779cf9878a979f41953470a7 (2021-11-17 13:25:49 -0500)
web: add the location to button click analytics
also, clean up some test warnings

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: web/src/ApiButton.tsx [hunk 1]
```diff
@@ -169,6 +169,9 @@ function ApiButtonWithOptions(props: ApiButtonWithOptionsProps & ButtonProps) {
   const { submit, uiButton, setInputValue, getInputValue, ...buttonProps } =
     props
 
+  let componentType = uiButton.spec?.location?.componentType
+  let tags = { component: componentType }
```
Comment (lizzthabet) [inline web/src/ApiButton.tsx:173]: wdyt about adding the `annotations` property to the tags? i'm thinking about how we'd want to distinguish between user-defined buttons vs. tilt-defined buttons, like the enable / disable toggle. (this would introduce a dependency between this and #5182, so it'd be fine if you wanted to do that in a separate pr!)

**Review by lizzthabet — APPROVED** [review]
i'm excited to have button analytics ahead of enable/disable launching. 🌟

File: web/src/ApiButton.tsx (hunk 1 above)
Comment (nicks) [reply web/src/ApiButton.tsx:173]: i think that's a good idea! but we'd have to be a little careful to make sure we don't accidentally include personally-identifiable data (like if the annotation contains something about the user's project)

Comment (lizzthabet) [reply web/src/ApiButton.tsx:173]: oh yeah! maybe we could just allowlist the known tilt annotations and ignore everything else.
